Add diligent academy coding challenge (#2)
DiligentCorp/rc-tech-assessment: diligent-academy-assessments/backend/coding-challenge/README.md

# Coding challenge

## How to use

> - You can run the tests using the command `npm test` to check your work, they are running in watch mode, meaning you do
>   not need to rerun them when you change your code.
> 

Reviewer: ```suggestion
- Use `npm test` to check your work. This command runs the tests in watch mode, so you do  not need to rerun it when you change your code.

```

## Principles and rules

- You can change code in the `src/implementations` folder.
- **Do not modify anything in the `src/sealed` and `test` folders.**

## Tasks

Implement the methods in the `user.js` file.

The `User` class has the following methods:

- `borrow(book, library)` - This method should add the book to the user's borrowed books list, and
  remove the book from the library's available books list.

- `hasBook(book)` - This method should return a boolean if the user has the book borrowed.

> - `return(book library)` - This method should remove the book from the user's borrowed books list,

Reviewer: ```suggestion
- `return(book, library)` - This method should remove the book from the user's borrowed books list,
```

  and
  add the book to the library's available books list. It should return a boolean if it could return the book

>   because the user had it borrowed.

Reviewer: Do we only return with a `boolean` value if the book is borrowed? Should we say the we return with a `true` value if we successfully returned the book and `false` otherwise?

Author: Oh yeah good idea, thank you, the original TS version was much clearer on this. :D
